platform/solr/solr-factory-impl/src/main/java/org/codice/solr/factory/impl/Configsets.java

```diff
@@ -80,11 +84,61 @@ private Path createDefaultConfigset(Path configsets) {
 
   public Path get(String collection) {
     Path collectionPath = Paths.get(configsetsPath, collection, "conf");
-    if (collectionPath.toFile().exists()) {
-      return collectionPath;
-    } else {
+    if (!collectionPath.toFile().exists()) {
       LOGGER.debug("No configset for collection [{}]. Using default configset instead", collection);
       return defaultPath;
     }
+    return getRectifiedConfig(collection, collectionPath);
+  }
+
+  private Path getRectifiedConfig(String collection, Path collectionPath) {
+    // if any defaults are missing-- create a temp  and copy collectionPath + defaults
+    Set<Path> allFiles =
+        SOLR_CONFIG_FILES.stream()
+            .map(configName -> Paths.get(collectionPath.toString(), configName))
+            .collect(Collectors.toSet());
+    Set<Path> missingFiles =
+        allFiles.stream().filter(config -> !config.toFile().exists()).collect(Collectors.toSet());
+    Set<Path> existingFiles =
+        allFiles.stream().filter(missingFiles::contains).collect(Collectors.toSet());
+
+    if (missingFiles.isEmpty()) {
+      return collectionPath;
+    }
+
+    Path tempConfigDir;
+    try {
+      tempConfigDir = Files.createTempDirectory("solr-configsets");
```

Comment:
❓ Shouldn't we include the collection name in the temporary folder so two clients creating different collections do not partially overwrite each other?

Reply:
Never mind.  The directory will have a random number added to it.

https://github.com/AdoptOpenJDK/openjdk-jdk8u/blob/5ca4d2a8ca68334a9f7323ebf9f5a259cee6f6ff/jdk/src/share/classes/java/nio/file/TempFileHelper.java#L57-L65